### ISSUE TITLE
System.InvalidOperationException: 'Stream cannot be read. Please send us the PDF file so that we can fix this'

### ISSUE THREAD

**Issue by robrechtD** [empira/PDFsharp#148]
Opening a pdf from a stream produces the following exception: 

System.InvalidOperationException: 'Stream cannot be read. Please send us the PDF file so that we can fix this (issues (at) pdfsharp.net).'
I've attached a project to reproduce the issue.

[PDFsharp.IssueSubmissionTemplate.zip](https://github.com/user-attachments/files/16560301/PDFsharp.IssueSubmissionTemplate.zip)

**Comment by ThomasHoevel** [comment]
Thanks for the feedback.
The PDF file is corrupted in object 14 0.
The object has an empty stream, but the length is given as 2338 bytes and that's more bytes than there are remaining in the file.
```
14 0 obj
<</Length 2338
/Type
/Metadata
/Subtype 
/XML>>
stream
endstream
```
We try to improve PDFsharp with respect to reading corrupted files, but I cannot say if or when a workaround for that corruption will come.

**Comment by packdat** [comment]
@ThomasHoevel
I just added a commit to #152 that handles this situation.

**Comment by ThomasHoevel** [comment]
Issue should be resolved with PDFsharp 6.2.0 Preview 2.
